### ISSUE TITLE
[🐛 Bug]: Certain element methods (`isExisting`, `getText`) don't work in `browser.switchFrame`, others don't work in `browser.switchToFrame`

### ISSUE THREAD

**Issue by FluffyJay1** [webdriverio/webdriverio#13808]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

9.2.1

### Node.js Version

20.18.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
Chrome, Firefox, Windows, Linux
```

### What happened?

When working with elements in an iframe, depending on the method used to enter the iframe context, certain element methods will not return the expected results. For example:
|   | browser.switchToFrame | browser.switchFrame |
| - | ---- | ---- |
| element.isExisting() | ❌ (always returns false) | ✅ |
| element.getText()  | ✅ | ❌ (always returns an error object) | 

I have not tested other methods but I imagine the root issue has to do with the protocols used to implement these methods. 

### What is your expected behavior?

At least `browser.switchFrame` should work with all element commands like `element.getText()`.

### How to reproduce the bug.

Minimal reproduction example: https://github.com/FluffyJay1/wdio-iframe-elements

### Relevant log output

```typescript
For `browser.switchFrame`:

[0-0] isExisting: true
[0-0] So isExisting() works on that element... but what about getText()
[0-0] getText(): {"error":"no such element","message":"no such element: element not found\n  (Session info: chrome=130.0.6723.70)","stacktrace":"\tGetHandleVerifier [0x00007FF64F0C3A95+28005]\n\t(No symbol) [0x00007FF64F028390]\n\t(No symbol) [0x00007FF64EEC580A]\n\t(No symbol) [0x00007FF64EED7D0B]\n\t(No symbol) [0x00007FF64EECCF04]\n\t(No symbol) [0x00007FF64EECAE09]\n\t(No symbol) [0x00007FF64EECE5E7]\n\t(No symbol) [0x00007FF64EECE6A0]\n\t(No symbol) [0x00007FF64EF0F9DB]\n\t(No symbol) [0x00007FF64EF3BA3A]\n\t(No symbol) [0x00007FF64EF09246]\n\t(No symbol) [0x00007FF64EF3BC50]\n\t(No symbol) [0x00007FF64EF5B8B3]\n\t(No symbol) [0x00007FF64EF3B7E3]\n\t(No symbol) [0x00007FF64EF075C8]\n\t(No symbol) [0x00007FF64EF08731]\n\tGetHandleVerifier [0x00007FF64F3B641D+3118829]\n\tGetHandleVerifier [0x00007FF64F406C70+3448640]\n\tGetHandleVerifier [0x00007FF64F3FCEED+3408317]\n\tGetHandleVerifier [0x00007FF64F18A3EB+841403]\n\t(No symbol) [0x00007FF64F0333EF]\n\t(No symbol) [0x00007FF64F02F464]\n\t(No symbol) [0x00007FF64F02F5FD]\n\t(No symbol) [0x00007FF64F01EB59]\n\tBaseThreadInitThunk [0x00007FF927F07374+20]\n\tRtlUserThreadStart [0x00007FF928D3CC91+33]\n"}
[6504:25724:1028/003959.514:ERROR:device_event_log_impl.cc(201)] [00:39:59.513] USB: usb_service_win.cc:105 SetupDiGetDeviceProperty({{A45C254E-DF1C-4EFD-8020-67D146A850E0}, 6}) failed: Element not found. (0x490)
[0-0] Error in "should see element text when using browser.switchFrame"
Error: Bug
Expect  to have text

Expected: StringContaining "unpartitioned-cookie"
Received: {"error": "no such element", "message": "no such element: element not found
...
```

For `browser.switchToFrame`:
```
[0-1] getText(): __Host-partitioned-cookie=123; unpartitioned-cookie=456
[0-1] So getText() works on that element... but what about isExisting()
[0-1] isExisting: false
[30008:24036:1028/004008.552:ERROR:device_event_log_impl.cc(201)] [00:40:08.551] USB: usb_service_win.cc:105 SetupDiGetDeviceProperty({{A45C254E-DF1C-4EFD-8020-67D146A850E0}, 6}) failed: Element not found. (0x490)
[0-1] Error in "should see element exists when using browser.switchToFrame"
Error: Bug
Expect $(`#cookies`) to be existing

Expected: "existing"
Received: "not existing"
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by danielhjacobs** [comment]
One minute before you posted this, WebdriverIO 9.2.2 released with #13781, which may have fixed the `switchFrame` behavior.

**Comment by christian-bromann** [comment]
@danielhjacobs is right here we just pushed out an improvement for this. We do not recommend to use `switchToFrame` anymore as it only switches frames for WebDriver Classic commands. We have created `switchFrame` which behaves the same (and has more capabilities). So switching the frame via `await browser.switchFrame($('iframe'));` works for both Chrome and Firefox.

I will go ahead and close this but let me know if you still encounter any issues.
